### PR TITLE
fix(connlib): forward PTR queries for non-resources

### LINKED ISSUE #6749: `Unknown resource dst=100.100.111.1` in trace logs
I noticed `**.google.com` isn't working for me as a Resource. `firezone-prod` Gateways (GCP). Upon attempting to debug, the Gateway logs look fine:

```
Sep 18 04:32:06 gateway-7zhj sudo[1701]: 2024-09-18T04:32:06.591055Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
Sep 18 04:32:06 gateway-7zhj sudo[1701]: 2024-09-18T04:32:06.742648Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
Sep 18 04:32:06 gateway-7zhj sudo[1701]: 2024-09-18T04:32:06.806975Z  INFO handle_timeout{cid=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b}: snownet::node: Updating remote socket old=None new=Relay { relay: 9df94599-2e5a-4f7a-b06c-dbc7cf44ad41, dest: 107.197.104.68:58889 } duration_since_intent=0ns
Sep 18 04:32:06 gateway-7zhj sudo[1701]: 2024-09-18T04:32:06.845409Z  INFO handle_timeout{cid=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b}: snownet::node: Updating remote socket old=Some(Relay { relay: 9df94599-2e5a-4f7a-b06c-dbc7cf44ad41, dest: 107.197.104.68:58889 }) new=Direct { source: [2600:1900:4020:4c:0:74::]:60672, dest: [2600:1700:3ecb:2410:3d11:5d4e:79e8:73c2]:52694 } duration_since_intent=250ms
Sep 18 04:32:06 gateway-7zhj sudo[1701]: 2024-09-18T04:32:06.898685Z  INFO snownet::node: Completed wireguard handshake cid=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b duration_since_intent=307.752357ms
Sep 18 04:32:07 gateway-7zhj sudo[1701]: 2024-09-18T04:32:07.570671Z  INFO snownet::allocation: Updated allocation srflx_ip4=Some(35.237.252.155:54596) srflx_ip6=Some([2600:1900:4020:4c:0:74::]:60672) relay_ip4=Some(34.48.110.184:55720) relay_ip6=Some([2600:1900:4091:51f:0:1a::]:55720) remaining_lifetime=Some(600s)
Sep 18 04:32:08 gateway-7zhj sudo[1701]: 2024-09-18T04:32:08.513796Z  INFO snownet::allocation: Updated allocation srflx_ip4=Some(35.237.252.155:54596) srflx_ip6=Some([2600:1900:4020:4c:0:74::]:60672) relay_ip4=Some(34.23.106.0:49748) relay_ip6=Some([2600:1900:4020:8f68:0:6b::]:49748) remaining_lifetime=Some(600s)
Sep 18 04:32:27 gateway-7zhj sudo[1701]: 2024-09-18T04:32:27.476172Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
Sep 18 04:32:33 gateway-7zhj sudo[1701]: 2024-09-18T04:32:33.283071Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
Sep 18 04:32:49 gateway-7zhj sudo[1701]: 2024-09-18T04:32:49.303652Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
Sep 18 04:32:50 gateway-7zhj sudo[1701]: 2024-09-18T04:32:50.947879Z  INFO firezone_tunnel::gateway: Allowing access to resource client=91b9f3e6-7b9c-465f-aca8-9f4d7e8c4e3b resource=95ab56ae-619f-4a12-8b3b-48b6004f559c expires=Some("2024-09-25T04:32:01+00:00")
```

And the Client debug logs look fine, but I noticed **lots** of these messages when using `firezone_tunnel=trace,debug`:

```
connlib	21:34:35.675359-0700	firezone_tunnel::client: Unknown resource dst=100.100.111.1
connlib	21:34:35.675412-0700	firezone_tunnel::dns: Parsed packet as DNS query: 'PTR 225.1.168.192.in-addr.arpa'
connlib	21:34:35.675457-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=**.ifconfig.co id=431da4dd-f88b-4a7f-9f4c-3673f98dfe2e
connlib	21:34:35.675616-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=**.google.com id=95ab56ae-619f-4a12-8b3b-48b6004f559c
connlib	21:34:35.675663-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=metabase.c.firezone-prod.internal id=9d0a0e7a-6a86-4498-a8e3-c19ff9677fdb
connlib	21:34:35.675774-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=showmyip.com id=47f613b6-6a52-4e17-a88c-5aa06d1438a9
connlib	21:34:35.675894-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=ifconfig.net id=0a00a5eb-a03e-4010-82ad-c7d4c2738517
connlib	21:34:35.675931-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=dnsleaktest.com id=a7510aba-e731-4f0a-aaec-102b705712fd
connlib	21:34:35.675966-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=speed.cloudflare.com id=fec99d5a-8537-4d04-829d-55ab1fcf875c
connlib	21:34:35.676074-0700	match_resource_linear{domain=225.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No resources matched
connlib	21:34:35.676145-0700	firezone_tunnel::client: Unknown resource dst=100.100.111.1
connlib	21:34:39.857629-0700	firezone_tunnel::dns: Parsed packet as DNS query: 'PTR b._dns-sd._udp.0.1.168.192.in-addr.arpa'
connlib	21:34:39.857781-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=**.ifconfig.co id=431da4dd-f88b-4a7f-9f4c-3673f98dfe2e
connlib	21:34:39.857846-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=**.google.com id=95ab56ae-619f-4a12-8b3b-48b6004f559c
connlib	21:34:39.857892-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=metabase.c.firezone-prod.internal id=9d0a0e7a-6a86-4498-a8e3-c19ff9677fdb
connlib	21:34:39.857939-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=showmyip.com id=47f613b6-6a52-4e17-a88c-5aa06d1438a9
connlib	21:34:39.858164-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=ifconfig.net id=0a00a5eb-a03e-4010-82ad-c7d4c2738517
connlib	21:34:39.858211-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=dnsleaktest.com id=a7510aba-e731-4f0a-aaec-102b705712fd
connlib	21:34:39.858674-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No match pattern=speed.cloudflare.com id=fec99d5a-8537-4d04-829d-55ab1fcf875c
connlib	21:34:39.858721-0700	match_resource_linear{domain=b._dns-sd._udp.0.1.168.192.in-addr.arpa}: firezone_tunnel::dns: No resources matched
connlib	21:34:39.858841-0700	firezone_tunnel::client: Unknown resource dst=100.100.111.1
```

This results in all `google.com` domains failing to load (in browsers only!):

Screenshot 2024-09-17 at 9 35 26 PM

### PR DESCRIPTION
**Pull request by thomaseizinger**
When encountering a PTR query, `connlib` checks if the query is for a Firezone-managed resource and resolve it to the correct IP. If it isn't for a DNS resource, we should forward the query to the upstream resolver.

This isn't what is currently happening though. Instead of forwarding the query, we bail early from `StubResolver::handle` and thus attempt to route the packet through the tunnel. This however fails because the DNS query was targeted at `connlib`'s stub resolver address which never corresponds to a resource IP.

When TRACE logs where activated, this resulted in several entries such as

> Unknown resource dst=100.100.111.1

To ensure this doesn't regress, we now generate PTR and MX record queries in `tunnel_test`. We don't assert the response of those but we do assert that we always get a response. The inclusion of MX records asserts that unknown query types get correctly forwarded.

Resolves: #6749.

### CODE REVIEW

**Review by conectado — APPROVED** [review]
Good catch! This could be causing issues with DNS in a lot of cases

**Comment by thomaseizinger** [comment]
> This isn't what is currently happening though. Instead of forwarding the query, we bail early from `StubResolver::handle` and thus attempt to route the packet through the tunnel. This however fails because the DNS query was targeted at `connlib`'s stub resolver address which never corresponds to a resource IP.

I'll send a follow-up where we I want to refactor the function signature to explicitly track the cases of "packet for our DNS resolver but failed otherwise".

**Comment by thomaseizinger** [comment]
I ran 10000 proptest cases locally. Appears to be stable :)

File: rust/connlib/tunnel/src/tests/transition.rs [hunk 1]
```diff
@@ -186,28 +191,49 @@ pub(crate) fn dns_queries(
...
             })
             .collect::<Vec<_>>()
     })
 }
 
-pub(crate) fn intercepted_query_type() -> impl Strategy<Value = Rtype> {
-    prop_oneof![Just(Rtype::A), Just(Rtype::AAAA)]
+fn ptr_query_ip() -> impl Strategy<Value = IpAddr> {
+    prop_oneof![
+        host_v4(IPV4_RESOURCES).prop_map_into(),
```
Comment (thomaseizinger) [inline rust/connlib/tunnel/src/tests/transition.rs:224]: @conectado It just occurred to me that the chances of hitting one of these with a resource that we actually resolved are incredibly low. We should choose a much smaller netmask here (like 30 maybe?) to ensure we also generate some for the first IPs that we assign for DNS resources.

**Pull request merged**